src/_pytest/python.py

```diff
@@ -1230,6 +1231,14 @@ def idmaker(
         _idvalset(valindex, parameterset, argnames, idfn, ids, config=config, item=item)
         for valindex, parameterset in enumerate(parametersets)
     ]
+    # rewrite parametrized ids greater than a 100 limit
```

Comment:
we have to do this for all ids and we should use the auto-id mechanism for that + warn

it has to happen on all platforms, else users get different test ids on different platforms thus destroying reasonably simple aggregation

Reply:
Thanks @RonnyPfannschmidt for your time, i'm struggling to warn here without is causing a collection error, any advice?

Reply:
at first glance i would propose to use the `node.warn` apis if applicable, and to move the limit into the `_idval` helper function in order to get a normal numbered id like other tests with non-repressentable ids

Reply:
@RonnyPfannschmidt any advice here? i am struggling to warn here with warnings.warn() / item/warn() ? `node.warn` I am not too familiar with.  Attempting to warning is just erroring,  I assume due to collection/interpretation.

Reply:
ill at best be able to look into this mid week as im currently preoccupied